### PR TITLE
[OS Patch] TensorFlow 2.9

### PR DESCRIPTION
**Pull request by junpuf**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by junpuf** [comment]
#2268

**Comment by junpuf** [comment]
#2270

File: tensorflow/inference/docker/2.9/py3/Dockerfile.cpu [hunk 1]
```diff
@@ -97,8 +97,9 @@ RUN ${PIP} install --no-cache-dir \
     gevent==21.12.0 \
     requests==2.27.1 \
     grpcio==1.46.3 \
-    protobuf==3.19.4 \
-# using --no-dependencies to avoid installing tensorflow binary
+    # Protobuf 3.18.3, 3.19.5, 3.20.2 and 4.21.6 include a fix for CVE-2022-1941
+    protobuf==3.19.5 \
```
Comment (saimidu) [inline tensorflow/inference/docker/2.9/py3/Dockerfile.cpu:101]: nit: Could you please change this to `"protobuf>=3.19.5,<3.20"` so that it won't require updates in the future if protobuf were to come out with a new patch version?

File: tensorflow/inference/docker/2.9/py3/cu112/Dockerfile.gpu [hunk 2]
```diff
@@ -143,8 +143,9 @@ RUN ${PIP} install -U --no-cache-dir \
     gevent==21.12.0 \
     requests==2.27.1 \
     grpcio==1.46.3 \
-    protobuf==3.19.4 \
-# using --no-dependencies to avoid installing tensorflow binary
+    # Protobuf 3.18.3, 3.19.5, 3.20.2 and 4.21.6 include a fix for CVE-2022-1941
+    protobuf==3.19.5 \
```
Comment (saimidu) [inline tensorflow/inference/docker/2.9/py3/cu112/Dockerfile.gpu:147]: nit: Could you please change this to `"protobuf>=3.19.5,<3.20"` so that it won't require updates in the future if protobuf were to come out with a new patch version?

File: tensorflow/training/docker/2.9/py3/Dockerfile.cpu [hunk 3]
```diff
@@ -159,7 +159,8 @@ RUN ${PIP} install --no-cache-dir -U \
     "werkzeug>=2.0.2,<3" \
     psutil==5.7.2 \
     "urllib3>1.26.0<1.27.0" \
-    protobuf==3.19.4
+    # Protobuf 3.18.3, 3.19.5, 3.20.2 and 4.21.6 include a fix for CVE-2022-1941
+    protobuf==3.19.5
```
Comment (saimidu) [inline tensorflow/training/docker/2.9/py3/Dockerfile.cpu:163]: nit: Could you please change this to `"protobuf>=3.19.5,<3.20"` so that it won't require updates in the future if protobuf were to come out with a new patch version?

File: tensorflow/training/docker/2.9/py3/cu112/Dockerfile.gpu [hunk 4]
```diff
@@ -222,7 +222,8 @@ RUN ${PIP} install --no-cache-dir -U \
    "opencv-python>=4.3,<5" \
    "werkzeug>=2.0.2,<3" \
    "urllib3>1.26.0<1.27.0" \
-   protobuf==3.19.4
+    # Protobuf 3.18.3, 3.19.5, 3.20.2 and 4.21.6 include a fix for CVE-2022-1941
+    protobuf==3.19.5
```
Comment (saimidu) [inline tensorflow/training/docker/2.9/py3/cu112/Dockerfile.gpu:226]: nit: Could you please change this to `"protobuf>=3.19.5,<3.20"` so that it won't require updates in the future if protobuf were to come out with a new patch version?

**Review by saimidu — APPROVED** [review]

File: dlc_developer_config.toml [hunk 5]
```diff
@@ -23,7 +23,7 @@ benchmark_mode = false
 [build]
 # Frameworks for which you want to disable both builds and tests
 # available frameworks - ["autogluon", "huggingface_tensorflow", "huggingface_pytorch", "huggingface_tensorflow_trcomp", "huggingface_pytorch_trcomp", "tensorflow", "mxnet", "pytorch"]
-skip_frameworks = []
+skip_frameworks = ["pytorch", "mxnet", "autogluon", "huggingface_tensorflow", "huggingface_pytorch", "huggingface_tensorflow_trcomp", "huggingface_pytorch_trcomp"]
```
Comment (saimidu) [inline dlc_developer_config.toml:26]: Please revert these changes for a merge.

File: tensorflow/inference/docker/2.9/py3/Dockerfile.cpu (hunk 1 above)
Comment (junpuf) [reply tensorflow/inference/docker/2.9/py3/Dockerfile.cpu:101]: consumed the changes from master

File: tensorflow/training/docker/2.9/py3/Dockerfile.cpu (hunk 3 above)
Comment (junpuf) [reply tensorflow/training/docker/2.9/py3/Dockerfile.cpu]: consumed the changes from master

File: dlc_developer_config.toml (hunk 5 above)
Comment (junpuf) [reply dlc_developer_config.toml]: reverted

**Review by kace — DISMISSED** [review]
LGTM

**Review by xncqr — APPROVED** [review]
LGTO

File: test/dlc_tests/sanity/test_safety_check.py [hunk 6]
```diff
@@ -76,6 +78,8 @@
                 "43453",
                 # tensorflow-estimator and tensorflow versions must match. For all TF versions below TF 2.9.0, we cannot upgrade tf-estimator to 2.9.0
                 "48551",
+                # for cryptography until e have 39.0.0 release
```
Comment (arjkesh) [inline test/dlc_tests/sanity/test_safety_check.py:81]: nit: typo, e --> we

File: test/dlc_tests/sanity/test_safety_check.py [hunk 7]
```diff
@@ -55,6 +55,8 @@
                 "42814",
                 # tensorflow-estimator and tensorflow versions must match. For all TF versions below TF 2.9.0, we cannot upgrade tf-estimator to 2.9.0
                 "48551",
+                # for cryptography until e have 39.0.0 release
```
Comment (arjkesh) [inline test/dlc_tests/sanity/test_safety_check.py:58]: nit: typo, e --> we

**Review by arjkesh — APPROVED** [review]
